From 28896a6654e48d76c07ba21dba32bc616f1c43f8 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Thu, 4 Jun 2015 23:26:19 +0530
Subject: [PATCH] [USB_TYPEC_PD]: Set platform pd capability based on fusb phy
 version id

When setting the platform PD capability check for the fusb phy version
id. If the fusb phy version id is for fusb302 chip, then only set the
platform is pd cable.

Change-Id: I01c439926436c89c57c7d322b68b10745e1de2df
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38557
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 62683f022ab5..83a2caae8cd2 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -1200,7 +1200,11 @@ err_measure:
 
 static bool fusb300_pd_capable(struct typec_phy *phy)
 {
-	if (phy->type == USB_TYPE_C)
+	struct fusb300_chip *chip;
+
+	chip = dev_get_drvdata(phy->dev);
+	/* set the paltform is pd capable only if the phy is fusb302 */
+	if (phy->type == USB_TYPE_C && !chip->is_fusb300)
 		return true;
 	else
 		return false;
-- 
1.9.1

